### ISSUE TITLE
False positive with no-obj-calls

### ISSUE THREAD

**Issue by julien-f** [eslint/eslint#12437]
**Tell us about your environment**

```
Environment Info:

Node version: v8.16.1
npm version: v6.11.3
Local ESLint version: v6.5.1 (Currently used)
Global ESLint version: Not found
```

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: { es6: true },
  rules: { "no-obj-calls": "error" }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var construct = typeof Reflect !== "undefined" ? Reflect.construct : undefined;
construct();
```

```bash
eslint index.js
```

**What did you expect to happen?**

Should produce no error because I'm not calling `Reflect`.

**What actually happened? Please include the actual, raw output from ESLint.**

```
/index.js
  2:1  error  'construct' is not a function  no-obj-calls

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by mdjermanovic** [comment]
Hi @julien-f , thanks for the report!

Confirmed, and there are other false positives: [Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLW9iai1jYWxsczogW1wiZXJyb3JcIl0gKi9cblxudmFyIGNvbnN0cnVjdCA9IHR5cGVvZiBSZWZsZWN0ICE9PSBcInVuZGVmaW5lZFwiID8gUmVmbGVjdC5jb25zdHJ1Y3QgOiB1bmRlZmluZWQ7XG5jb25zdHJ1Y3QoKTtcblxudmFyIGEgPSBSZWZsZWN0ID8gMSA6IDI7XG5hKCk7XG5cbnZhciBiID0gdHJ1ZSA/IDEgOiBSZWZsZWN0O1xuYigpO1xuXG52YXIgYyA9IGZhbHNlID8gUmVmbGVjdCA6IDE7XG5jKCk7XG5cbnZhciBkID0gdHJ1ZSB8fCBSZWZsZWN0O1xuZCgpOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6eyJlczYiOnRydWV9fX0=)

@mysticatea could this be a problem in `eslint-utils`, or maybe the rule doesn't use `ReferenceTracker` well?

**Comment by mysticatea** [comment]
This seems to be a bug in `eslint-utils` package. I guess the problem is in [src/reference-tracker.js#L229-L232](https://github.com/mysticatea/eslint-utils/blob/4e1bc077c2a6bb00538d66b69a63c24de3463bed/src/reference-tracker.js#L229-L232).
